src/librustc/hir/intravisit.rs

```diff
@@ -279,6 +279,12 @@ pub trait Visitor<'v> : Sized {
     fn visit_ty(&mut self, t: &'v Ty) {
         walk_ty(self, t)
     }
+    fn visit_ty_param(&mut self, t: &'v TyParam) {
```

Comment:
This isn't actually used - I'd prefer if people went through `visit_generic_param` or `visit_generics` instead - that way, when const generics are added, they're harder to miss.

Also, have you considered removing `visit_lifetime_def`? I believe that was the reason I mentioned changing `Vec<LifetimeDef>` to `Generics`, because then all the cases can be handled in one place in all visitors.

Reply:
Do you mean inlining just `visit_ty_param`, or `walk_ty_param` too?